LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -6,172 +6,246 @@ namespace LearningMissionSimulation
-                accountDictionary.Add(account.Id, account);
+                AccountDictionary.Add(account.Id, account);
+
+                ReportItem(itemName: account.ToString(), actionName: action, itemIndex: i);
+
                 if (account.Role == Role.Student)
                 {
                     Student student = ObjectGenerator.GenerateStudent(account.Id);
-                    studentList.Add(student);
+                    StudentDictionary.Add(account.Id, student);
                     if (account.Status == Status.Pending)
                     {
-                        pendingAccountQueue.Enqueue(account);
+                        PendingAccountQueue.Enqueue(account);
+                    }
+                    else if(account.Status == Status.Active)
+                    {
+                        ActiveStudentList.Add(student);
                     }
                 }
                 else if(account.Role == Role.Instructor)
                 {
                     Instructor instructor = ObjectGenerator.GenerateInstructor(account.Id);
-                    instructorList.Add(instructor);
+                    InstructorDictionary.Add(account.Id, instructor);
                     if(account.Status == Status.Pending)
                     {
-                        pendingAccountQueue.Enqueue(account);
+                        PendingAccountQueue.Enqueue(account);
+                    }
+                    else if (account.Status == Status.Active)
+                    {
+                        ActiveInstructorList.Add(instructor);
                     }
                 }
                 i++;
             }
-            ReportSummary("Generated", "accounts", accountCount);
-            ReportFooter("Create Account");
+
+            ReportSummary(actionName: action, itemCount: accountCount);
+            ReportFooter(actionName: action);
         }
 
         public void ActivateAccounts()
         {
-            ReportHeader("Activate account");
-            int pendingAccountQueueCount = pendingAccountQueue.Count;
+            string action = "Activate account";
+            ReportHeader(actionName: action);
+
+            int pendingAccountQueueCount = PendingAccountQueue.Count;
             int i = 0;
-            while(pendingAccountQueue.Count > 0)
+            while(PendingAccountQueue.Count > 0)
             {
-                Account account = pendingAccountQueue.Dequeue();
+                Account account = PendingAccountQueue.Dequeue();
                 account.Status = Status.Active;
-                ReportItem(account.ToString(), "Activated account", i++);
+                if(account.Role == Role.Student)
+                {
+                    Student student = null;
+                    if (StudentDictionary.Count != 0)
+                    {
+                        StudentDictionary.TryGetValue(account.Id, out student);
+                    }
+                    ActiveStudentList.Add(student);
```

Comment:
ActiveStudentList.Add(student) should be inside the **if** block. You should not add nulls to the list